docs/Guides/time_dependent.rst

```diff
@@ -54,3 +54,8 @@ Therefore a total of 42 customers passed through the system::
 
    >>> len(Q.nodes[-1].all_individuals)
    42
+
+Time dependent function can be used for batching distributions too.
+In this case the time dependent function must return the number of individuals that must enter the simulation at the given time.
+Note that the function must return a float number, but the result will then be rounded to the nearest integer.
```

Comment:
I believe that you are saying this (and subsequently using `round` and `int` in the `batch_size` method of the ArrivalNode) because the `check_timedependent_dist` (https://github.com/CiwPython/Ciw/blob/master/ciw/simulation.py#L71) requires a float.

I believe it would be neater to sample integers directly. Therefore we can change the `check_timedependent_dist` method to be:

```
    def check_timedependent_dist(self, func, kind, current_time):
        sample = func(current_time)
        if kind in ['Arr', 'Ser'] and not isinstance(sample, float) or sample < 0:
            raise ValueError("TimeDependent func must return positive float.")
        if kind in ['Bch'] and not isinstance(sample, int) or sample < 0:
            raise ValueError("TimeDependent batching func must return positive integer.")
        return sample
```

Thus for arrival and service time distributions we sample floats only, and for batching distributions we sample integers only.

For this to work `check_timedependent_dist` must take in a `kind`, so line 145 of `simulation.py` (https://github.com/CiwPython/Ciw/blob/master/ciw/simulation.py#L145) would need to change to:

```
return lambda t : self.check_timedependent_dist(self.source(c, n, kind)[1], kind, t)
```